src/resources.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState, useCallback } from 'react';
+import http, { CancelToken } from 'axios';
+import { useAuthentication } from './authentication';
+
+import { fetchUser, updateUser, User } from './api';
+
+export function useUser(id: string) {
+  const [user, setUser] = useState<User | null>(null);
+  const { token } = useAuthentication();
+
+  useEffect(() => {
+    const cancelToken = http.CancelToken.source();
+
+    const loadUser = async (cancelToken?: CancelToken) => {
+      if (token) {
+        const fetchedUser = await fetchUser(id, { token, cancelToken });
+        setUser(fetchedUser);
+      } else {
+        setUser(null);
+      }
+    };
+
+    loadUser(cancelToken.token);
+    return () => cancelToken.cancel();
+  }, [id, token, setUser]);
+
+  const update = useCallback(
+    async (user: User) => {
```

Comment:
Similar thing here.  Would be much easier to understand the file, if you gave this a name which describes what it does.

Reply:
This will, when called, update the user remotely and then update the local state of the user. I'd just call it `updateUser`, but as it is returned from `useUser`, it's implied you're updating the user, which is why I just called it `update`.

Also actually, the `useCallback` is no longer needed, we could give the callback directly. I originally was using this function inside another effect, which meant I needed to make sure its identity didn't change. I'd like to keep this behaviour, but it is a bit of a premature optimization as it's no longer directly used.